lib/client/debug/debug.go

```diff
@@ -0,0 +1,152 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package debug
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/gravitational/trace"
+
+	apidefaults "github.com/gravitational/teleport/api/defaults"
+	"github.com/gravitational/teleport/lib/srv/debug"
+)
+
+// Client specifies the debug service client.
+type Client interface {
+	// SetLogLevel changes the application's log level and a change status message.
+	SetLogLevel(context.Context, string) (string, error)
+	// GetLogLevel fetches the current log level.
+	GetLogLevel(context.Context) (string, error)
+	// CollectProfile collects a pprof profile.
+	CollectProfile(context.Context, string, int) ([]byte, error)
+}
+
+type clt struct {
+	clt *http.Client
+}
+
+// NewClient generates a new debug service client.
+func NewClient(socketPath string) Client {
+	return &clt{
+		clt: &http.Client{
+			Timeout: apidefaults.DefaultIOTimeout,
+			Transport: &http.Transport{
+				DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+					return net.Dial("unix", socketPath)
+				},
+			},
+		},
+	}
+}
+
+// SetLogLevel changes the application's log level and a change status message.
+func (c *clt) SetLogLevel(ctx context.Context, level string) (string, error) {
+	resp, err := c.do(ctx, http.MethodPut, url.URL{Path: debug.LogLevelEndpoint}, []byte(level))
```

Comment:
It was ok to source `debug.LogLevelEndpoint` from tools but now it feels wrong to source server lib from client lib. Could we move the constants to some common place? Sorry for the back and forth on the constants.

Reply:
Moved the constants to `constants.go`.